packages/flutter/lib/src/widgets/semantics_debugger.dart

```diff
@@ -250,22 +250,24 @@ class _SemanticsDebuggerPainter extends CustomPainter {
     final SemanticsData data = node.getSemanticsData();
     final List<String> annotations = <String>[];
 
-    bool wantsTap = false;
-    if (data.hasFlag(SemanticsFlag.hasCheckedState)) {
-      annotations.add(data.hasFlag(SemanticsFlag.isChecked) ? 'checked' : 'unchecked');
-      wantsTap = true;
-    }
-    if (data.hasFlag(SemanticsFlag.isTextField)) {
-      annotations.add('textfield');
-      wantsTap = true;
+    if (data.hasFlag(SemanticsFlag.isHeader)) {
+      annotations.add('header');
     }
 
-    if (data.hasAction(SemanticsAction.tap)) {
-      if (!wantsTap)
-        annotations.add('button');
-    } else {
-      if (wantsTap)
+    if (data.hasFlag(SemanticsFlag.hasCheckedState)) {
+      annotations.add(data.hasFlag(SemanticsFlag.isChecked) ? 'checked' : 'unchecked');
+      if (!data.hasAction(SemanticsAction.tap)) {
```

Comment:
Instead of looking for the tap Action, we should probably look for the presence of the SemanticsFlags hasEnabledState/isEnabled to show if the element is disabled or not. That's what talkback/voiceover use to actually announce something as enabled/disabled.

Reply:
Yup, that is better. I hadn't noticed the enabled state in the data. Thanks for the feedback.